### PR TITLE
Create usecase action references validator

### LINKED ISSUE #646: Create Use Case Action References Validator
As an AaC Developer, I want a use case action validator, so that I know I have valid use case action references to behaviors. 

## AC:

- [x] A new validator is created for use case action references
- [x] A unit test for the validator is created
- [x] The validator is added to the pipeline core validators if necessary
- [x] The validator documentation is updated to cover the new validator
- [x] Actions in use cases are defined behaviors in the source (participant model)

### PR DESCRIPTION
**Pull request by cvchaparro**
# Description

Add a validator that ensures all of the `action` fields in `usecase` steps refer to a valid behavior in the `step`'s `source` model.

# Linked Items:

Closes #646 

### Added

- A new validator which validates the `action` field of each of a `usecase`'s `steps`.

### Changed

- Clean up whitespace from `spec.yaml`.

# Checklist:

- [x] I updated project documentation to reflect my changes.
- [x] My changes generate no new warnings.
- [x] I updated new and existing unit tests to account for my changes.
- [x] I linked the associated item(s) to be closed.
- [x] I bumped the version.

### CODE REVIEW

File: python/tests/plugins/first_party/test_gen_design_doc_impl.py [hunk 1]
```diff
@@ -163,11 +164,7 @@ def assert_use_case(self, markdown: str) -> None:
     - step: move an object from one place to another
       source: model1
       target: model2
-      action: move from point alpha to point beta
-    - step: move an object back to it's original location
-      source: model2
-      target: model1
-      action: move from point beta back to point alpha
+      action: {TEST_BEHAVIOR_NAME}
```
[the remark this answers is not in the document]
Comment (cvchaparro) [reply python/tests/plugins/first_party/test_gen_design_doc_impl.py:167]: This comment applies to all of your questions: The reason I replaced it this way is to make sure the name of the behavior will always match with the source model's available behavior. This isn't technically required but it makes dealing with long behavior names slightly less error-prone in tests.

Hope that clarifies it but let me know if you have any more questions.

[the remark this answers is not in the document]
Comment (lizzcondrey) [reply python/tests/plugins/first_party/test_gen_design_doc_impl.py:167]: So it isn't problematic that the replacement constant doesn't always match the string it replaced? That was my primary concern. The ones where it did, totally understood the swap to the constant. But the ones that don't match, like `ProcessDataA` getting replaced by `TEST_PUML_BEHAVIOR_ONE_NAME` when they didn't match.

[the remark this answers is not in the document]
Comment (lizzcondrey) [reply python/tests/plugins/first_party/test_gen_design_doc_impl.py:167]: Or for where this comment resides, how `action` and the `step` got to be removed and just replaced with the new `action` reference.

[the remark this answers is not in the document]
Comment (lizzcondrey) [reply python/tests/plugins/first_party/test_gen_design_doc_impl.py:167]: Discussed in person what these change outs meant and why they came about.

**Review by Coffee2Bits — APPROVED** [review]
LGTMoving, Cameron. 🚀

**Pull request merged**